--- src/decompiled/Terraria/Initializers/AssetInitializer.cs
+++ src/NTerraria/Terraria/Initializers/AssetInitializer.cs
@@ -14,6 +_,7 @@
 using Terraria.GameContent.UI;
 using Terraria.ID;
 using Terraria.IO;
+using Terraria.NTerraria.Core;
 using Terraria.Utilities;
 
 namespace Terraria.Initializers
@@ -591,7 +_,8 @@
 			LoadMinimapFrames(mode);
 			LoadPlayerResourceSets(mode);
 			Main.AchievementAdvisor.LoadContent();
-		}
+            NTTextureAssets.Initialize(mode);
+        }
 
 		private static void LoadMinimapFrames(AssetRequestMode mode) {
 			float num = 2f;
@@ -649,6 +_,6 @@
 			obj.Put("PlayerResourcesSet", text);
 		}
 
-		private static Asset<T> LoadAsset<T>(string assetName, AssetRequestMode mode) where T : class => Main.Assets.Request<T>(assetName, mode);
+		internal static Asset<T> LoadAsset<T>(string assetName, AssetRequestMode mode) where T : class => Main.Assets.Request<T>(assetName, mode);
 	}
 }
